docs/contributing/03-stories/story-16.md

```diff
@@ -0,0 +1,100 @@
+# User Story 16 - Wing for TypeScript
+
+> **Status**: Draft
+
+Anders loves TypeScript, but he also loves the amazing experience Wing offers for building cloud applications.
+
+It's Andres' lucky day because Wing Cloud had just announced that very soon TypeScript developers are able to start leveraging Wing libraries from the comfort of
+their favorite programming language.
+
+He got an invite for a pre-release. The invite explains that initially, there is only support for `cloud.Bucket` and `cloud.Function` and for consuming [winglibs](https://github.com/winglang/winglibs).
+
+The example looks like this:
+
+First, install some libraries:
+
+```sh
+$ npm i winglang wingsdk @winglibs/cloud @winglibs/checks
+```
+
+```ts
+// main.ts
+import { inflight, App } from "wingsdk";
+import * as cloud from "@winglibs/cloud";
+import { Check } from "@winglibs/checks";
+
+// Final App depends on platform used
+const app = new App();
+
+const data = "world";
+const bucket = new cloud.Bucket(app, "Bucket");
+
+const fn = new cloud.Function(app, "Function", inflight({ bucket, data }, async (ctx, event) => {
+  await ctx.bucket.put("hello.txt", ctx.data);
+});
+
+bucket.grantPut(fn);
+
+const check = new Check(app, "Check", inflight({ message, fn, bucket, data }, async (ctx, event) => {
+  await fn.invoke();
+  const actual = await bucket.get("hello.txt");
+
+  console.log(`actual = ${actual}, expected = ${data}`);
+
+  if (actual !== ctx.data) {
+    throw new Error("check failed");
+  }
+}));
+
+fn.grantInvoke(check);
+bucket.grantGet(check);
+
+app.synth();
+```
+
+Now, you can open the Wing Console:
+
+```sh
+$ wing run main.ts
+```
+
+And compile to AWS:
+
+```sh
+$ wing compile -t tf-aws main.ts
+```
+
+And enjoy life!
+
+## Notes
+
+The `@winglibs/cloud` library is a Wing library that looks like this:
+
+```js
+bring cloud;
```

Comment:
I know what you mean here, although technically `bring cloud` also means "bring cloud from wingsdk".

I guess the compiler should read `cloud` here as always referring to the internal one first, so I assume this won't be a problem

Reply:
Good point. The intention here *is* to refer to the standard library (internal) module and not to the winglib.

But as you say, since in winglang the internal modules take precedence, then in TypeScript this kind of makes sense.

I am wondering if eventually this is where we will put the code of the SDK (once we start writing it in Wing).

Reply:
I'll use this thread to talk about a circular dependency that I've been mentioning.

Currently, The implementations of `std.Test` use `cloud.Function`. `std.Test` is built into wing and is effectively a language feature, so it makes sense that it would have to stay in an internal library. If `@winglang/sdk` no longer had `cloud.Function`, this internal library would depend on `@winglibs/cloud` instead. This means the compiler itself would not be able to compile `@winglibs/cloud` (or at best it could not use `test "" {}`)

If `@winglibs/cloud` is a simple proxy to `@winglang/sdk` then this isn't an issue, but it's a hurdle to think about later.

Reply:
> If @winglibs/cloud is a simple proxy to @winglang/sdk then this isn't an issue, but it's a hurdle to think about later.

Cool :-)